src/matchers.ts

```diff
@@ -56,3 +62,146 @@ export function isAllOf<Matchers extends [Matcher<any>, ...Matcher<any>[]]>(
     return matchers.every(matcher => matches(matcher, action))
   }
 }
+
+/**
+ * @param action A redux action
+ * @param validStatus An array of valid meta.requestStatus values
+ *
+ * @internal
+ */
+export function hasExpectedRequestMetadata(action: any, validStatus: string[]) {
+  if (!action || !action.meta) return false
+
+  const hasValidRequestId = typeof action.meta.requestId === 'string'
+  const hasValidRequestStatus =
+    validStatus.indexOf(action.meta.requestStatus) > -1
+
+  return hasValidRequestId && hasValidRequestStatus
+}
+
+export type UnknownAsyncThunkPendingAction = ReturnType<
+  AsyncThunkPendingActionCreator<unknown>
+>
+
+/**
+ * A type guard function that matches any pending async thunk action.
+ *
+ * @param action A redux action
+ *
+ * @public
+ */
+export function isPending(
+  action: any
+): action is UnknownAsyncThunkPendingAction {
+  return hasExpectedRequestMetadata(action, ['pending'])
+}
+
+export type UnknownAsyncThunkRejectedAction = ReturnType<
+  AsyncThunkRejectedActionCreator<unknown, unknown>
+>
+
+/**
+ * A type guard function that matches any rejected async thunk action.
+ *
+ * @param action A redux action
+ *
+ * @public
+ */
+export function isRejected(
+  action: any
+): action is UnknownAsyncThunkRejectedAction {
+  return hasExpectedRequestMetadata(action, ['rejected'])
+}
+
+export type UnknownAsyncThunkFulfilledAction = ReturnType<
+  AsyncThunkFulfilledActionCreator<unknown, unknown>
+>
+
+/**
+ * A type guard function that matches any fulfilled async thunk action.
+ *
+ * @param action A redux action
+ *
+ * @public
+ */
+export function isFulfilled(
+  action: any
+): action is UnknownAsyncThunkFulfilledAction {
+  return hasExpectedRequestMetadata(action, ['fulfilled'])
+}
+
+export type UnknownAsyncThunkAction =
+  | UnknownAsyncThunkPendingAction
+  | UnknownAsyncThunkRejectedAction
+  | UnknownAsyncThunkFulfilledAction
+
+/**
+ * A type guard function that matches any async thunk action.
+ *
+ * @param action
+ *
+ * @internal
+ */
+export function isAnyAsyncThunkAction(
```

Comment:
I think this could be part of `isAsyncThunkAction`, when `isAsyncThunkAction` is called with only one argument that is not of type function.

Reply:
I don't mind making the change, but I wonder if it's more efficient to reuse the same function?  Also, pulling it out and giving it a name may make the code easier for a reader to understand.  What do you think?

Reply:
It's actually not too bad if I move `isAnyAsyncThunkAction` inside `isAsyncThunkAction`.

```ts
  if (asyncThunks.length === 0) {
    return (action: any) => hasExpectedRequestMetadata(action, [
      'pending',
      'fulfilled',
      'rejected'
    ])
  }
```

I'll include that change shortly.